h11/_headers.py

```diff
@@ -74,6 +74,11 @@ def normalize_and_validate(headers):
         # downstream." -- https://tools.ietf.org/html/rfc7230#section-3.2.4
         if name.strip() != name:
             raise LocalProtocolError("Illegal header name {}".format(name))
+        # "A sender MUST NOT generate a message that includes line folding"
+        # -- https://tools.ietf.org/html/rfc7230#section-3.2.4
```

Comment:
That's not really the relevant section -- it's saying that when encoding headers into the wire format, you shouldn't use line folding as part of the encoding. The rule we're enforcing here is that the header payload (the thing that gets encoded -- think like unicode strings versus byte strings) cannot contain `\r` or `\n` (among other things).